javascript/code-challenges/challenges-01.test.js

```diff
@@ -104,10 +119,20 @@ The inventory is formatted like this:
 This function should use forEach to populate your grocery list based on the store's inventory. If the item is available, add it to your list. Return the final list.
 ------------------------------------------------------------------------------------------------ */
 
+let arr2 =[];
```

Comment:
this is where you're running into problems.
More your ```let arr2 = [ ];
inside of your function. So this could read
const createList = (availableItems) => {
  let arr2=[ ];
  availableItems.forEach... etc etc

Reply:
I see this change (inserting array in function) will let the test run, and show it as correct.  But, outside the testing environment, why will my solution be wrong?  If my array2 is unique, and only runs with that function, then it should be a viable solution - correct?